lib/migration.js

```diff
@@ -138,15 +127,105 @@ function mixinMigration(MsSQL) {
       }
     });
 
-    // drop columns
-    actualFields.forEach(function(f) {
-      var notFound = !~propNames.indexOf(f.Field);
-      if (f.Field === idName) return;
-      if (notFound || !m.properties[f.Field]) {
-        columnsToDrop.push(self.columnEscaped(model, f.Field));
+    if (columnsToAdd.length) {
+      statements.push('ADD ' + columnsToAdd.join(',' + MsSQL.newline));
+    }
+
+    if (columnsToAlter.length) {
+      // SQL Server doesn't allow multiple columns to be altered in one statement
+      columnsToAlter.forEach(function(c) {
+        statements.push('ALTER COLUMN ' + c);
+      });
+    }
+
+    function actualize(propName, oldSettings) {
+      var newSettings = m.properties[propName];
+      if (newSettings && changed(newSettings, oldSettings)) {
+        columnsToAlter.push(self.columnEscaped(model, propName) + ' ' +
+          self.propertySettingsSQL(model, propName));
+      }
+    }
+
+    function changed(newSettings, oldSettings) {
+      if (oldSettings.Null === 'YES' &&
+        (newSettings.allowNull === false || newSettings.null === false)) {
+        return true;
       }
+      if (oldSettings.Null === 'NO' && !(newSettings.allowNull === false ||
+        newSettings.null === false)) {
+        return true;
+      }
+      if (oldSettings.Type.toUpperCase() !== datatype(newSettings)) {
+        return true;
+      }
+      return false;
+    }
+    return statements;
+  };
+
+  MsSQL.prototype.getDropColumns = function(model, actualFields) {
+    var sql = [];
+    var self = this;
+    sql = sql.concat(self.getColumnsToDrop(model, actualFields));
+    return sql;
+  };
+
+  MsSQL.prototype.getColumnsToDrop = function(model, actualFields) {
+    var self = this;
+    var m = this._models[model];
+    var propNames = Object.keys(m.properties);
+    var idName = this.idName(model);
+
+    var statements = [];
+    var columnsToDrop = [];
+
+    if (actualFields) {
+      // drop columns
+      actualFields.forEach(function(f) {
+        var notFound = !~propNames.indexOf(f.Field);
+        if (f.Field === idName) return;
+        if (notFound || !m.properties[f.Field]) {
+          columnsToDrop.push(self.columnEscaped(model, f.Field));
+        }
+      });
+
+      if (columnsToDrop.length) {
+        statements.push('DROP COLUMN' + columnsToDrop.join(',' + MsSQL.newline));
+      }
+    };
+    return statements;
+  };
+
+  MsSQL.prototype.addIndexes = function(model, actualIndexes, cb) {
```

Comment:
Does it need a cb? it's not called inside function and the caller doesn't provide it either.